### PR TITLE
Add contexts to traits from floras

### PR DESCRIPTION
**Pull request by ehwenk**
For many of the traits from the floras, especially the numeric traits, there is a column `category` that captures the exact plant part to which the measurement refers (e.g. leaf vs leaflet). There were 100+ distinct values in this column, many of which are just slight syntax differences.

This is a first pass at condensing these into more curated context values. It is difficult, because all values that distinguish between replicate measurements for a single taxon by trait combination should be maintained, but terms that aren't being used for this purpose should be collapsed.

There are also a few instances where there truly are multiple values for a single taxon by trait combination within a flora and these can then be assigned "replicate observation" numbers, allowing the datasets to pivot wider.

Currently, they all pivot, but this does not mean all information that could be mapped as a context has been. It is just a realistic ending point for now.

### CODE REVIEW

File: data/ABRS_2023/metadata.yml [hunk 1]
```diff
@@ -32,13 +32,71 @@ dataset:
...
+        entity_measured = ifelse(category %in% c("lateral_leaf", "lateral_leaves", "axillary_shoot_leaves", "axillary_stem_leaves"), "lateral_leaf", entity_measured),
+        entity_measured = ifelse(category %in% c("cauline_leaves", "cauline_lamina", "caulineleaves", "cauline_leaf"), "cauline_leaf", entity_measured),
+        entity_measured = ifelse(category %in% c("basal_leaves", "radical_leaves", "basal_eaves", "nest_fronds", "radical_leaf", "basal_lamina","basal_leaf", "rosette_leaves"), "basal_leaf", entity_measured),
+        entity_measured = ifelse(category %in% c("phyllode", "phyllodes"), "phyllode", entity_measured),
+        entity_measured = ifelse(category %in% c("cladode", "cladodes", "phylloclades"), "cladode", entity_measured),
+        entity_measured = ifelse(category %in% c("leafsegments"), "lobes", entity_measured),
+        entity_measured = ifelse(category %in% c("lobes", "stem_leaf", "leaf"), "all", entity_measured),
```
Comment (dcol2804) [inline data/ABRS_2023/metadata.yml:79]: Does this line mean that lobes, stem_leaf or leaf categories will be replaced by "all"?

File: data/ABRS_2023/metadata.yml [hunk 2]
```diff
@@ -32,13 +32,71 @@ dataset:
         trait_value = ifelse(trait_name == "leaf_shape", stringr::str_replace(trait_value, "cylindrical", "terete"), trait_value),
         trait_value = ifelse(trait_name == "leaf_margin_posture", stringr::str_replace(trait_value, "recurved", "revolute"), trait_value),
         trait_value = ifelse(trait_name == "leaf_margin_posture", stringr::str_replace(trait_value, "incurved", "involute"), trait_value),
-        measurement_remarks = ifelse(is.na(category), NA, paste0("Trait value refers to ", category))
-      )
+        measurement_remarks = ifelse(is.na(category), NA, paste0("Trait value refers to ", category)),
```
Comment (dcol2804) [inline data/ABRS_2023/metadata.yml:35]: Wow! Fantastic work and very impressed you got it to pivot

File: data/ABRS_2023/metadata.yml (hunk 1 above)
Comment (ehwenk) [reply data/ABRS_2023/metadata.yml:79]: > Does this line mean that lobes, stem_leaf or leaf categories will be replaced by "all"?

Yes, that was in one of your scripts, but maybe I was meant to change it? Feel free to adjust anything.

**Review by yangsophieee — APPROVED** [review]
I wasn't able to check the details of collapsing the categories but I skimmed the metadata changes and it looks good :)

**Pull request merged**